src/chunk.c

```diff
@@ -2766,18 +2766,29 @@ init_scan_by_compressed_chunk_id(ScanIterator *iterator, int32 compressed_chunk_
 }
 
 bool
-ts_chunk_contains_compressed_data(Chunk *chunk)
+ts_chunk_contains_compressed_data(Chunk *chunk, Chunk **parent)
```

Comment:
Why not  add a new function which returns the parent chunk instead of overloading this one? That would keep the semantics and usage separate.

Reply:
I can do it, just not a fan of doing the same scan twice